hnn_core/tests/test_network.py

```diff
@@ -50,6 +64,20 @@ def test_network():
     print(network_builder)
     print(network_builder._cells[:2])
 
+    # Ensure distant dependent calcium gbar
+    gid = net.gid_ranges['L5_pyramidal'][0]
+    for section_name, section in network_builder._cells[gid].sections.items():
+        ca_gbar = list()
+        for seg in section.allseg():
+            try:  # Necessary for seg end points with no calcium mech
```

Comment:
avoid `try`/`except` when you can. It can hide issues. If you really must, you should limit the `except` clause to a class of exceptions you expect. Like `except ValueError` etc. You can probably use `hasattr` or just do `section.allseg()[1:-1]`